pydevices/HtsDevices/acq2106_WRTD.py

```diff
@@ -0,0 +1,148 @@
+            {'path':':WR_INIT:WRTD_RX_M',   'type':'text', 'value': "acq2106_999", 'options':('write_shot',)},
+            {'path':':WR_INIT:WRTD_RX_M1',  'type':'text', 'value': "acq2106_999", 'options':('write_shot',)},
+            {'path':':WR_INIT:WRTD_RX_DTP', 'type':'text', 'value': "acq2106_999", 'options':('write_shot',)},
+            {'path':':WR_INIT:WRTD_DELAY',  'type':'numeric', 'value': 5000000, 'options':('write_shot',)},   # From uut.cC.WRTD_DELAY01
+            {'path':':WR_INIT:WRTD_ID',     'type':'text', 'value': "acq2106_999", 'options':('write_shot',)},
+            {'path':':WR_INIT:WRTD_TX',     'type':'numeric', 'value': 0, 'options':('write_shot',)},
+            {'path':':WR_INIT:WRTD_RX',     'type':'numeric', 'value': 0, 'options':('write_shot',)},
+
+        {'path':':RUNNING',     'type':'numeric', 'options':('no_write_model',)},
+        {'path':':LOG_OUTPUT',  'type':'text',    'options':('no_write_model', 'write_once', 'write_shot',)},
+        {'path':':INIT_ACTION', 'type':'action',  'valueExpr':"Action(Dispatch('CAMAC_SERVER','INIT',50,None),Method(None,'INIT',head))", 'options':('no_write_shot',)},
+        {'path':':STOP_ACTION', 'type':'action',  'valueExpr':"Action(Dispatch('CAMAC_SERVER','STORE',50,None),Method(None,'STOP',head))",'options':('no_write_shot',)},
+        ]
+
+    def init(self, newmsg=''):
+        import acq400_hapi
+        uut = acq400_hapi.Acq2106(self.node.data(), has_wr=True)
+
+        if ":" in newmsg:
+            msgs  = str(newmsg)
+            wrmgs = msgs.split(":")
+            print("Messages are {} and {}".format(str(wrmgs[0]), str(wrmgs[1])))
+        else:
+            raise Exception("Argument needs to be of the following format: [d0msg1, d0msg2,]:[d1msg1, d1msg2,]")
+
+        #Record the state of the WRTD environment:
+        self.wr_init_wrtd_rx_m.record   = str(wrmgs[0]) # Matches for WRTT0
+        self.wr_init_wrtd_rx_m1.record  = str(wrmgs[1]) # Matches for WRTT1
+
+        # WR TRIGGER SOURCE:
+        print("WR TRG Source {}".format(str(self.trig_src.data())))
+        uut.s0.WR_TRG_DX = str(self.trig_src.data())
+
+        # Global WR settings:
+        # Set WRTD_ID
+        print("WRTD_ID {}".format(str(self.wr_init_wrtd_id.data())))
+        uut.cC.WRTD_ID = str(self.wr_init_wrtd_id.data())
+
+        # Sets WR "safe time for broadcasts" the message
+        uut.cC.WRTD_DELTA_NS=self.wr_init_wrtd_dns.data()
+
+        # Receiver:
+        # Turn on RX
+        uut.cC.WRTD_RX = int(self.wr_init_wrtd_rx.data())
+        # Define RX matches
+        uut.cC.WRTD_RX_MATCHES  = str(wrmgs[0])
+        uut.cC.WRTD_RX_MATCHES1 = str(wrmgs[1])
+        #Commit the changes for WRTD RX
+        uut.cC.wrtd_commit_rx = 1
+
+        # Transmiter:
+        # Turn on TX
+        uut.cC.WRTD_TX = int(self.wr_init_wrtd_tx.data())
+        #Commit the changes for WRTD TX
+        uut.cC.wrtd_commit_tx = 1
+
+    INIT=init
+
+    def trig(self, msg=''):
```

Comment:
Can you explain the trig method ?  Is this 'listen for trig'?  If so. do we want a repeat ? (probably don't need it) do we want a cancel ?

Reply:
Josh, so trigger has two ways of using it: with or without an argument.
Without arguments, sets the trigger so that it can be hardware triggered or trigger by a particular message in the WR network.
With an argument, a message is sent immediately. If the systems are armed and WR configuration ready, then the message will trigger them.